nixos/modules/services/web-servers/nginx/vhost-options.nix

```diff
@@ -199,13 +199,43 @@ with lib;
     };
 
     http3 = mkOption {
+      type = types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Whether to enable HTTP/3 protocol.
+        This requires using `pkgs.nginxQuic` package
+        which can be achieved by setting `services.nginx.package = pkgs.nginxQuic;`
+        and activate the QUIC transport protocol
+        `services.nginx.virtualHosts.<name>.quic = true;`.
+        Note that HTTP/3 support is experimental and
+        *not* yet recommended for production.
+        Read more at https://quic.nginx.org/
+      '';
+    };
+
+    http3_hq = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to enable HTTP/0.9 protocol negotiation used in QUIC interoperability tests.
```

Comment:
http3_hq and http/0.9?

Reply:
Yes, it says so in the Readme - http://hg.nginx.org/nginx-quic/log/tip/README?patch=&amp;linerange=314:315#3c33d39a51d3-l1.124